src/Diagnostics.DataProviders/KustoHeartBeatService/KustoHeartBeatService.cs

```diff
@@ -107,75 +113,97 @@ public KustoHeartBeat(string primaryCluster, string failoverCluster, KustoDataPr
+        {
+            while (!cancellationToken.IsCancellationRequested) {
+                
+                string activityId = Guid.NewGuid().ToString();
 
-                if (primaryHeartBeatSuccess)
+                if (string.IsNullOrEmpty(FailoverCluster))
                 {
-                    _ConsecutiveFailureCount = 0;
-                    _ConsecutiveSuccessCount++;
+                    await RunHeartBeatPrimary(activityId);
                 }
                 else
                 {
-                    _ConsecutiveFailureCount++;
-                    _ConsecutiveSuccessCount = 0;
-                }
-
-                // if not in failover state
-                //  should failover?
-                if (UsePrimaryCluster && _ConsecutiveFailureCount >= _configuration.HeartBeatConsecutiveFailureLimit)
-                {
-                    UsePrimaryCluster = false;
-                } // else should stop failover
-                else if (!UsePrimaryCluster && _ConsecutiveSuccessCount >= _configuration.HeartBeatConsecutiveFailureLimit)
-                {
-                    UsePrimaryCluster = true;
+                    var primaryTask = RunHeartBeatPrimary(activityId);
+                    var failoverTask = RunHeartBeatFailover(activityId);
+                    await Task.WhenAll(new Task[] { primaryTask, failoverTask });
                 }
-
-                LogHeartBeatInformation(primaryHeartBeatSuccess, failoverHeartBeatSuccess, PrimaryCluster, FailoverCluster, activityId, _ConsecutiveSuccessCount, _ConsecutiveFailureCount, exceptionForLog);
-
                 await Task.Delay(TimeSpan.FromSeconds(_configuration.HeartBeatDelay), cancellationToken);
             }
         }
 
-        private void LogHeartBeatInformation(bool primaryClustersuccess, bool failoverClusterSuccess, string cluster, string failoverCluster, string invocationId, int successCount, int failureCount, Exception exception)
+        private void LogHeartBeatInformation(string primaryOrFailover, bool clustersuccess, string cluster, string activityId, bool usingPrimaryCluster, Exception exception)
         {
-            var primaryClusterStatus = primaryClustersuccess ? "Success" : "Failed";
-            var failoverClusterStatus = failoverClusterSuccess ? "Success" : "Failed";
-
+            var clusterStatus = clustersuccess ? "Success" : "Failed";
 
             DiagnosticsETWProvider.Instance.LogKustoHeartbeatInformation(
-               invocationId,
-               $"PrimaryClusterStatus:{primaryClusterStatus},FailoverClusterStatus:{failoverClusterStatus},PrimaryCluster:{cluster},FailoverCluster:{failoverCluster},PrimaryClusterConsecutiveSuccessCount:{successCount},PrimaryClusterConsecutiveFailureCount:{failureCount}",
+               activityId,
+               $"ClusterType:{primaryOrFailover},ClusterStatus:{clusterStatus},Cluster:{cluster},UsingPrimaryCluster:{usingPrimaryCluster}",
```

Comment:
Perhaps change `{clusterStatus}` to `clusterStatus ? "Success" : "Failed"`

Reply:
Are you saying to inline the conditional operation?
If so I think it looks cleaner outside the string as I don't have to escape the quotation marks.

Reply:
Ah sounds good okay